mono/utils/mono-threads-state-machine.c

```diff
@@ -27,13 +27,20 @@ get_thread_suspend_count (int thread_state)
 	return (thread_state & THREAD_SUSPEND_COUNT_MASK) >> THREAD_SUSPEND_COUNT_SHIFT;
 }
 
+static inline gboolean
+get_thread_wont_safept (int thread_state)
+{
+	return (thread_state & THREAD_SUSPEND_WONT_SAFEPT_MASK) >> THREAD_SUSPEND_WONT_SAFEPT_SHIFT;
+}
+
 static inline int
-build_thread_state (int thread_state, int suspend_count) 
+build_thread_state (int thread_state, int suspend_count, gboolean wont_safept) 
 {
 	g_assert (suspend_count >= 0 && suspend_count <= THREAD_SUSPEND_COUNT_MAX);
 	g_assert (thread_state >= 0 && thread_state <= STATE_MAX);
+	g_assert (wont_safept == 0 || wont_safept == 1);
```

Comment:
Nit: Assert kinda redundant with `gboolean`, but I understand, it isn't quite.
If we ever get C++ `bool`...

Reply:
Maybe `no_safepoints = !!no_safepoints; // ensure it's 0 or 1`?
I'm paranoid about bitmasks and `gboolean` not really being a single bit.

(also renaming wont_safept to no_safepoints as elsewhere)